circuit/src/models/trade.ts

```diff
@@ -0,0 +1,85 @@
+import { TradeSide } from "../types/TradeSide";
+import { ZkpParams } from "../types/ZkpParams";
+import { genRandomInt, getPoseidonHash } from "../utils/Crypto";
+import Univ2 from "./Univ2";
+
+export default class Trade {
+    pool: Univ2;
+    salt: bigint;
+    amountIn: bigint;
+    amountOut: bigint;
+    reserve_in_image: string;
+    reserve_out_image: string;
+    zkpParams: ZkpParams;
```

Comment:
looks like there's a lot of overlap between Trade class tributes and ZkpParams attributes

what do you think of unifying (only storing once)? to minimize code surface area